code/modules/projectiles/projectile.dm

```diff
@@ -649,6 +649,10 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 		return FALSE
 	if((proj.ammo.flags_ammo_behavior & AMMO_XENO) && (isnestedhost(src) || stat == DEAD))
 		return FALSE
+	if(HAS_TRAIT(src, TRAIT_ISINTRENCH))
+		if(prob(75)) //75% chance to MISS.
+			return FALSE
```

Comment:
projectile_hit is returning the accuracy already, so use it.

Reply:
Working on it.